### ISSUE TITLE
Try to reconnect when calling ConnectAsync()

### ISSUE THREAD

**Issue by PranKe01** [doghappy/socket.io-client-csharp#72]
It would be nice, when SocketIO tries to "reconnect", even when the server was not available on ConnectAsync(). Right now I have a while loop around the ConnectAsync function, because my app is started, before the socketIO server starts. Without the loop, I would get an exception and the app would never try again to connect.

**Comment by PranKe01** [comment]
I also noticed, that it will throw an exception, when the server is not reachable and we try to send via EmitAsync(). SocketIO itself contains a buffer/queue for those messages. It would be nice when this buffering mechanism is used, instead of throwing an exception. :)

**Comment by omercavuslu** [comment]
@doghappy any progress?

**Comment by doghappy** [comment]
Please upgrade to `v2.0.2.8`

```cs
            var client = new SocketIO("http://localhost:11001", new SocketIOOptions
            {
                AllowedRetryFirstConnection = true
            });
```

**Comment by PranKe01** [comment]
The AllowedRetryFirstConnection property is not available in netfx version 1.0.0.6?

**Comment by omercavuslu** [comment]
> Please upgrade to `v2.0.2.8`
> 
> ```cs
>             var client = new SocketIO("http://localhost:11001", new SocketIOOptions
>             {
>                 AllowedRetryFirstConnection = true
>             });
> ```
AllowedRetryFirstConnection working correctly but OnReconnecting event don't stable v2.0.2.8 version, it only hit one time when connection lost

**Comment by doghappy** [comment]
Hmmmm. 

I released a severely flawed version (v2.0.2.8), and I have fixed it.

Please check the Changelog:

- [SocketIOClient](https://github.com/doghappy/socket.io-client-csharp/blob/master/CHANGELOG.md)
- [SocketIOClient.NetFx](https://github.com/doghappy/socket.io-client-csharp/blob/master/CHANGELOG-NetFx.md)

At the same time, I modified the version numbers of `SocketIOClient` and `SocketIOClient.NetFx`, and both became v2.0.2.9
